packages/fbjs/src/unicode/UnicodeUtilsExtra.js

```diff
@@ -62,20 +61,22 @@ function getCodePointsFormatted(str) {
   return codePoints.map(formatCodePoint);
 }
 
-// We use this funky constructor instead of the object
+// We use this funky function instead of the object
 // constructor because JS object keys are always coerced
 // to string, but we want a map of int -> string pairs.
```

Comment:
Based on the comment, seems like coercion is the reason, though it shouldn't really matter AFAIK.